services/staging/rs_server_staging/processors.py

```diff
@@ -11,22 +11,950 @@
+            cluster (LocalCluster): Dask LocalCluster instance managing computation resources, used in local mode
+                If this is None, it means we are in cluster mode, and we should dynamically connect
+                to the Dask cluster for each job.
+        """
+        #################
+        # Locals
+        self.headers: Headers = credentials.headers
+        self.stream_list: list = []
+        #################
+        # Env section
+        self.catalog_url: str = os.environ.get(
+            "RSPY_HOST_CATALOG",
+            "http://127.0.0.1:8003",
+        )  # get catalog href, loopback else
+        #################
+        # Database section
+        self.job_id: str = str(uuid.uuid4())  # Generate a unique job ID
+        self.detail: str = "Processing Unit was queued"
+        self.progress: float = 0.0
+        self.tracker: tinydb.table.Table = db
+        self.create_job_execution()
+        #################
+        # Inputs section
+        self.item_collection: RSPYFeatureCollectionModel = input_collection
+        self.catalog_collection: str = collection
+        self.catalog_item_name: str = item
+        self.provider: str = provider
+        self.assets_info: list = []
+        self.tasks: list = []
+        # Lock to protect access to percentage
+        self.lock = tinydb_lock
+        # Tasks finished
+        self.tasks_finished = 0
+        self.logger = Logging.default(__name__)
+        self.cluster = cluster
+        self.catalog_bucket = os.environ.get("RSPY_CATALOG_BUCKET", "rs-cluster-catalog")
+
+    # Override from BaseProcessor, execute is async in RSPYProcessor
+    async def execute(self):  # pylint: disable=arguments-differ, invalid-overridden-method
+        """
+        Asynchronously execute the RSPY staging process, starting with a catalog check and
+        proceeding to feature processing if the check succeeds.
+
+        This method first logs the creation of a new job execution and verifies the connection to
+        the catalog service. If the catalog connection fails, it logs an error and stops further
+        execution. If the connection is successful, it initiates the asynchronous processing of
+        RSPY features.
+
+        If the current event loop is running, the feature processing task is scheduled asynchronously.
+        Otherwise, the event loop runs until the processing task is complete.
+
+        Returns:
+            dict: A dictionary containing the job ID and a status message indicating the job
+                has started.
+                Example: {"started": <job_id>}
+
+        Logs:
+            ProcessorStatus.CREATED: Logs the creation of a new processing job.
+            Error: Logs an error if connecting to the catalog service fails.
+
+        Raises:
+            None: This method doesn't raise any exceptions directly but logs errors if the
+                catalog check fails.
+        """
+        if not self.item_collection:
```

Comment:
Check also for features with no assets, there's nothing to do.

Reply:
Done